rflx/expression.py

```diff
@@ -558,7 +563,8 @@ def ada_expr(self) -> ada.Expr:
     def z3expr(self) -> z3.BoolRef:
         z3exprs = [t.z3expr() for t in self.terms]
         boolexprs = [t for t in z3exprs if isinstance(t, z3.BoolRef)]
-        assert len(z3exprs) == len(boolexprs)
+        if len(z3exprs) != len(boolexprs):
```

Comment:
Why not `if not all(isinstance(t, z3.BoolRef) for t in z3exprs)`? Also see follwing instances.

Reply:
As `boolexprs` is also used in the return statement, it should be faster to compare the lengths instead of computing `boolexprs` twice.

Reply:
I have missed that. In this case it makes sense.